packages/rocketchat-lib/server/lib/interceptDirectReplyEmails.js

```diff
@@ -0,0 +1,124 @@
+import IMAP from 'imap';
+
+class IMAPIntercepter {
+	constructor() {
+		this.imap = new IMAP({
+			user: RocketChat.settings.get('Direct_Reply_Username'),
+			password: RocketChat.settings.get('Direct_Reply_Password'),
+			host: RocketChat.settings.get('Direct_Reply_Host'),
+			port: RocketChat.settings.get('Direct_Reply_Port'),
+			debug: RocketChat.settings.get('Direct_Reply_Debug') ? console.log : false,
+			tls: !RocketChat.settings.get('Direct_Reply_IgnoreTLS'),
+			connTimeout: 30000,
+			keepalive: true
+		});
+	}
+
+	openInbox(Imap, cb) {
+		Imap.openBox('INBOX', false, cb);
+	}
+
+	start() {
+		const self = this;
+		const Imap = this.imap;
+
+		Imap.connect();
+		// On successfully connected.
+		Imap.once('ready', Meteor.bindEnvironment(() => {
+			if (Imap.state !== 'disconnected') {
+				self.openInbox(Imap, Meteor.bindEnvironment((err) => {
+					if (err) {
+						throw err;
+					}
+					// fetch new emails & wait [IDLE]
+					self.getEmails(Imap);
+
+					// If new message arrived, fetch them
+					Imap.on('mail', Meteor.bindEnvironment(() => {
+						self.getEmails(Imap);
+					}));
+				}));
+			} else {
+				console.log('IMAP didnot connected.');
+				Imap.end();
+			}
+		}));
+
+		Imap.once('error', function(err) {
+			console.log(err);
+			throw err;
+		});
+	}
+
+	isActive() {
+		const Imap = this.imap;
+		if (Imap.state === 'disconnected') {
```

Comment:
Use `if (this.imap.state === 'disconnected') {` there is no need to declare the variable above